cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -81,13 +81,13 @@ public void initialize(ClassLoadHelper loadHelper, SchedulerSignaler schedSignal
   @Override
   public void storeJob(JobDetail newJob, boolean replaceExisting) throws ObjectAlreadyExistsException {
     super.storeJob(newJob, replaceExisting);
-    executePersist(newJob, null);
+    executePersist(newJob, null, null);
   }
 
   @Override
   public void storeTrigger(OperableTrigger newTrigger, boolean replaceExisting) throws JobPersistenceException {
     super.storeTrigger(newTrigger, replaceExisting);
-    executePersist(null, newTrigger);
+    executePersist(null, newTrigger, super.getTriggerState(newTrigger.getKey()));
```

Comment:
Actually, looking at how this executePersist() is called, you can move the super.getTriggerState() into executePersist(). Because is it always called if the trigger is not null.